itests/kit/node_unmanaged.go

```diff
@@ -24,3 +60,730 @@ type TestUnmanagedMiner struct {
+
+	// Step 6: Submit the ProveCommit to the network
+	tm.t.Log("Submitting ProveCommitSector ...")
+
+	r = tm.submitMessage(ctx, &miner14.ProveCommitSectors3Params{
+		SectorActivations:        []miner14.SectorActivationManifest{{SectorNumber: sectorNumber}},
+		SectorProofs:             [][]byte{proveCommit},
+		RequireActivationSuccess: true,
+	}, 0, builtin.MethodsMiner.ProveCommitSectors3)
+	req.True(r.Receipt.ExitCode.IsSuccess())
+
+	tm.proofType[sectorNumber] = proofType
+
+	respCh := make(chan WindowPostResp, 1)
+
+	go tm.wdPostLoop(ctx, sectorNumber, respCh)
+
+	return sectorNumber, respCh
+}
+
+func (tm *TestUnmanagedMiner) wdPostLoop(ctx context.Context, sectorNumber abi.SectorNumber, respCh chan WindowPostResp) {
+	go func() {
+		currentEpoch, nextPost, err := tm.calculateNextPostEpoch(ctx, sectorNumber)
+		tm.t.Logf("Activating sector %d, next post %d, current epoch %d", sectorNumber, nextPost, currentEpoch)
+		if err != nil {
+			select {
+			case respCh <- WindowPostResp{SectorNumber: sectorNumber, Error: err}:
+			case <-ctx.Done():
+				return
+			default:
+			}
+			return
+		}
+
+		tm.FullNode.WaitTillChain(ctx, HeightAtLeast(nextPost))
+
+		err = tm.submitWindowPost(ctx, sectorNumber)
+		select {
+		case respCh <- WindowPostResp{SectorNumber: sectorNumber, Error: err}:
+		case <-ctx.Done():
+			return
+		default:
+		}
+	}()
+}
+
+func (tm *TestUnmanagedMiner) SubmitPostDispute(ctx context.Context, sectorNumber abi.SectorNumber) error {
+	tm.t.Logf("MinerB(%s): Starting dispute submission for sector %d", tm.ActorAddr, sectorNumber)
+
+	head, err := tm.FullNode.ChainHead(ctx)
+	if err != nil {
+		return fmt.Errorf("MinerB(%s): failed to get chain head: %w", tm.ActorAddr, err)
+	}
+
+	sp, err := tm.FullNode.StateSectorPartition(ctx, tm.ActorAddr, sectorNumber, head.Key())
+	if err != nil {
+		return fmt.Errorf("MinerB(%s): failed to get sector partition for sector %d: %w", tm.ActorAddr, sectorNumber, err)
+	}
+
+	di, err := tm.FullNode.StateMinerProvingDeadline(ctx, tm.ActorAddr, head.Key())
+	if err != nil {
+		return fmt.Errorf("MinerB(%s): failed to get proving deadline for sector %d: %w", tm.ActorAddr, sectorNumber, err)
+	}
+
+	disputeEpoch := di.Close + 5
+	tm.t.Logf("MinerB(%s): Sector %d - Waiting %d epochs until epoch %d to submit dispute", tm.ActorAddr, sectorNumber, disputeEpoch-head.Height(), disputeEpoch)
+
+	tm.FullNode.WaitTillChain(ctx, HeightAtLeast(disputeEpoch))
+
+	tm.t.Logf("MinerB(%s): Sector %d - Disputing WindowedPoSt to confirm validity at epoch %d", tm.ActorAddr, sectorNumber, disputeEpoch)
+
+	params := &miner14.DisputeWindowedPoStParams{
+		Deadline:  sp.Deadline,
+		PoStIndex: 0,
+	}
+
+	enc, aerr := actors.SerializeParams(params)
+	require.NoError(tm.t, aerr)
+
+	_, err = tm.FullNode.MpoolPushMessage(ctx, &types.Message{
+		To:     tm.ActorAddr,
+		From:   tm.OwnerKey.Address,
+		Value:  types.FromFil(1),
+		Method: builtin.MethodsMiner.DisputeWindowedPoSt,
+		Params: enc,
+	}, nil)
```

Comment:
`submitMessage()` should be good for this, right?

Reply:
Fixed this. Also made `submitMessage` propagate the error back to the caller.